File: examples/python/fastapi-sqlalchemy/repositories/organization.py
# =============================================================================
# Organization Repository (class-based)
# =============================================================================
# This example uses a class-based repository. The TypeScript example shows
# the same thing with free functions — both work equally well.

from sqlalchemy.orm import Session

from models import Organization


class OrganizationRepository:
    def __init__(self, session: Session):
        self.session = session

    def create(self, data: dict) -> dict:
        org = Organization(name=data["name"])
        self.session.add(org)
        self.session.commit()
        self.session.refresh(org)
        return {"id": org.id, "name": org.name}

    def delete(self, id: str) -> None:
        org = self.session.get(Organization, id)
        if org:
            self.session.delete(org)
            self.session.commit()

File: examples/typescript/express-prisma/src/repositories/organization.ts
// =============================================================================
// Organization Repository (free functions)
// =============================================================================
// This example uses plain exported functions. The Python example shows the
// same thing with a class-based repository — both work equally well.

import type { PrismaClient } from '@prisma/client'

export async function createOrganization(
  prisma: PrismaClient,
  data: { name: string },
) {
  return prisma.organization.create({ data: { name: data.name } })
}

export async function deleteOrganization(prisma: PrismaClient, id: string) {
  return prisma.organization.delete({ where: { id } })
}
